### ISSUE TITLE
Custom fields editable in the members portal

### ISSUE THREAD

**Issue by jefft** [tbar0970/jethro-pmm#996]
Within the `/members` portal, Jethro allows users to edit their own basic details:
![image](https://github.com/tbar0970/jethro-pmm/assets/205995/f502dcc8-bd05-473f-a595-a1760664b2ff)

There are some custom fields that it would be nice if users could update themselves. **WWCC** and **WWCC Expiry Date** are potential examples. For this it would be nice if custom fields could be flagged as editable by their owner. Something like:

![image](https://github.com/tbar0970/jethro-pmm/assets/205995/b8b5b828-2b46-4fde-b9be-9e5da65ffebf)

**Comment by gitanjo** [comment]
Hi Jeff. I like the direction of this and it will streamline admin if members can see and in some cases edit specific data fields. However, in our church the WWCC fields are managed by staff so there is no concern a member might pretend to have a WWCC or falsify an expiry date. So could we instead have two checkable options: 'Read-only by this member' and 'Self-editable by this member'.

**Comment by alburyAJ** [comment]
Hi Jeff. I like the direction of this and it will streamline admin if members can see and in some cases edit specific data fields. However for us, the WWCC fields are managed by staff so that no member need be concerned that someone could pretend to have a WWCC or modify an expiry date. So, could we perhaps have two checkable options: 'Read-only by this member' and 'Self-editable by this member'.

**Comment by tbar0970** [comment]
Yep, good idea, although WWCC is probably not a good example (it needs to be verified with OCG before being treated as legit)

**Comment by alburyAJ** [comment]
Agreed. A better self-editable example might be (say) 'My hobbies and interests' - a support to congregational connections but not a legal requirement. That said, if a field like WWCC could be made 'Read-only by all members', this might (say) help a kid's-outreach team to identify potential recruits, or help a parent confirm that a kid's-church helper is registered.
